test/jdk/java/lang/ref/FinalizerHistogramTest.java

```diff
@@ -32,37 +33,49 @@
  * @test
  * @summary Unit test for FinalizerHistogram
  * @modules java.base/java.lang.ref:open
- * @run main FinalizerHistogramTest
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm
+ *      -Xbootclasspath/a:.
+ *      -XX:+UnlockDiagnosticVMOptions
+ *      -XX:+WhiteBoxAPI
+ *      FinalizerHistogramTest
  */
 
 public class FinalizerHistogramTest {
     static ReentrantLock lock = new ReentrantLock();
-    static volatile int wasInitialized = 0;
-    static volatile int wasTrapped = 0;
-    static final int objectsCount = 1000;
+    static final AtomicInteger initializedCount = new AtomicInteger(0);
+    static final int OBJECTS_COUNT = 1000;
+
+    static WhiteBox wb;
 
     static class MyObject {
         public MyObject() {
             // Make sure object allocation/deallocation is not optimized out
-            wasInitialized += 1;
+            initializedCount.incrementAndGet();
         }
 
         protected void finalize() {
             // Trap the object in a finalization queue
-            wasTrapped += 1;
             lock.lock();
         }
     }
 
-    public static void main(String[] argvs) {
+    public static void main(String[] argvs) throws InterruptedException {
         try {
             lock.lock();
-            for(int i = 0; i < objectsCount; ++i) {
+            for(int i = 0; i < OBJECTS_COUNT; ++i) {
                 new MyObject();
             }
-            System.out.println("Objects intialized: " + objectsCount);
-            System.gc();
-            while(wasTrapped < 1);
+            System.out.println("Objects intialized: " + initializedCount.get());
+            wb = WhiteBox.getWhiteBox();
+            wb.fullGC();
+            boolean refProResult;
+            do {
+                refProResult = wb.waitForReferenceProcessing();
+                System.out.println("waitForReferenceProcessing returned: " + refProResult);
+            } while (refProResult);
```

Comment:
Why the spammy output?  Why not just
```
while (wb.waitForReferenceProcessing()) {}
```

Reply:
Right now, the test just it prints a single line:
`waitForReferenceProcessing returned: false`
and I expect this to continue to be true.

For intermittently failing tests, I'm inclined to add a little extra output as long as I'm fiddling with it anyway. That way, if the test ever starts failing again, we have some (hopefully useful) clues about what happened.

Reply:
> Right now, the test just it prints a single line: `waitForReferenceProcessing returned: false` and I expect this to continue to be true.

If reference processing is being slow because of load, that will be different.

> For intermittently failing tests, I'm inclined to add a little extra output as long as I'm fiddling with it anyway. That way, if the test ever starts failing again, we have some (hopefully useful) clues about what happened.

OK.